src/services/expenses.service.js

```diff
@@ -0,0 +1,86 @@
+let expenses = [];
+let nextId = 1;
+
+const resetExpenses = () => {
+  expenses = [];
+  nextId = 1;
+};
+
+const getExpenses = ({ userId, categoryList, from, to }) => {
+  return expenses.filter((expense) => {
+    let match = true;
+
+    if (userId && expense.userId !== +userId) {
+      match = false;
+    }
+
+    if (from && new Date(expense.spentAt) < new Date(from)) {
+      match = false;
+    }
+
+    if (to && new Date(expense.spentAt) > new Date(to)) {
+      match = false;
+    }
+
+    if (categoryList.length > 0 && !categoryList.includes(expense.category)) {
+      match = false;
+    }
+
+    return match;
+  });
+};
+
+const createExpense = ({ userId, spentAt, title, amount, category, note }) => {
```

Comment:
Consider adding validation for the input parameters to ensure they meet expected formats and constraints (e.g., `amount` should be a positive number, `spentAt` should be a valid date).